tracing-subscriber/src/filter/env/directive.rs

```diff
@@ -195,7 +195,7 @@ impl FromStr for Directive {
             )
             .unwrap();
             static ref SPAN_PART_RE: Regex =
-                Regex::new(r#"(?P<name>\w+)?(?:\{(?P<fields>[^\}]*)\})?"#).unwrap();
+                Regex::new(r#"(?P<name>[\w-]+)?(?:\{(?P<fields>[^\}]*)\})?"#).unwrap();
```

Comment:
I wonder if we should maybe just change this to 
```suggestion
                Regex::new(r#"(?P<name>[^\]\{]+)?(?:\{(?P<fields>[^\}]*)\})?"#).unwrap();
```
or something, so that we can accept any character in a span name except for the `]` that terminates the span filter or the `{` that signals the beginning of the span's fields?

This would allow filters to include span names with any arbitrary characters other than `]` and `{`, rather than just a fix for dashes in particular.

Of course, if we do this, it might be worthwhile to add more tests for span names with other characters in them, to make sure we don't do anything weird...